src/OrchardCore.Modules/OrchardCore.Users/ViewModels/RegisterExternalLoginViewModel.cs

```diff
@@ -1,65 +1,29 @@
-using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
-using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Localization;
-using OrchardCore.Email;
 
 namespace OrchardCore.Users.ViewModels
 {
-    public class RegisterExternalLoginViewModel : IValidatableObject
+    public class RegisterExternalLoginViewModel
     {
         public bool NoUsername { get; set; }
 
         public bool NoEmail { get; set; }
 
         public bool NoPassword { get; set; }
 
+        [Required(ErrorMessage = "Username is required.")]
         public string UserName { get; set; }
 
+        [Required(ErrorMessage = "Email is required.")]
+        [Email.EmailAddress(ErrorMessage = "Invalid Email.")]
         public string Email { get; set; }
 
+        [Required(ErrorMessage = "Passowrd is required.")]
         [DataType(DataType.Password)]
+        [StringLength(100, ErrorMessage = "Password must be between 6 and 100 characters.", MinimumLength = 6)]
         public string Password { get; set; }
 
+        [Compare(nameof(Password), ErrorMessage = "Confirm Password do not match.")]
         [DataType(DataType.Password)]
         public string ConfirmPassword { get; set; }
-
```

Comment:
From the below Validate(), I think you missed the logic that uses `NoUsername`, `NoEmail`, `NoPassword` properties.

Reply:
Ah, you're right I didn't notice that. I will check why do we need `NoEmail`, `NoUsername`